datasets/cbm_dataframe_parser.py

```diff
@@ -0,0 +1,392 @@
+
+        self.cnn_1dinput_len = cnn_1dinput_len
+
+        cnn_assert_message = "CNN input length is incorrect."
+        assert self.cnn_1dinput_len >= (self.target_sampling_rate_Hz *
+                                        self.signal_duration_in_sec)/2, cnn_assert_message
+
+        if not isinstance(self.downsampling_ratio, int) or self.downsampling_ratio < 1:
+            raise ValueError(
+                "downsampling_ratio can only be set to an integer value greater than 0"
+                )
+
+        processed_folder = \
+            os.path.join(root, self.__class__.__name__, 'processed')
+
+        self.processed_folder = processed_folder
+
+        makedir_exist_ok(self.processed_folder)
+
+        self.specs_identifier = f'eval_mode_{self.eval_mode}_' + \
+                                f'label_as_signal_{self.label_as_signal}_' + \
+                                f'ds_{self.downsampling_ratio}_' + \
+                                f'dur_{self.signal_duration_in_sec}_' + \
+                                f'ovlp_ratio_{self.overlap_ratio}_' + \
+                                f'random_split_{self.random_or_speed_split}_'
+
+        train_dataset_pkl_file_path = \
+            os.path.join(self.processed_folder, f'train_{self.specs_identifier}.pkl')
+
+        test_dataset_pkl_file_path =  \
+            os.path.join(self.processed_folder, f'test_{self.specs_identifier}.pkl')
+
+        if self.d_type == 'train':
+            self.dataset_pkl_file_path = train_dataset_pkl_file_path
+
+        elif self.d_type == 'test':
+            self.dataset_pkl_file_path = test_dataset_pkl_file_path
+
+        self.signal_list = []
+        self.lbl_list = []
+
+        self.__create_pkl_files()
+        self.is_truncated = False
+
+    def __create_pkl_files(self):
+        if os.path.exists(self.dataset_pkl_file_path):
+
+            print('\nPickle files are already generated ...\n')
+
+            (self.signal_list, self.lbl_list) = \
+                pickle.load(open(self.dataset_pkl_file_path, 'rb'))
+            return
+
+        self.__gen_datasets()
+
+    def normalize_signal(self, features):
+        """
+        Normalize signal with Local Min Max Normalization
+        """
+        # Normalize data:
+        for instance in range(features.shape[0]):
+            instance_max = np.max(features[instance, :, :], axis=1)
+            instance_min = np.min(features[instance, :, :], axis=1)
+
+            for feature in range(features.shape[1]):
+                for signal in range(features.shape[2]):
+                    features[instance, feature, signal] = (
+                        (features[instance, feature, signal] - instance_min[feature]) /
+                        (instance_max[feature] - instance_min[feature])
+                    )
+
+        return features
+
+    def __gen_datasets(self):
+
+        train_features = []
+        test_normal_features = []
+
+        train_speeds = []
```

Comment:
* Lets keep loads in a list as well
* Lets add to pickle both speed_list and load_list as well
* Lets add a speed_and_load_available parameter to init and only use it in the __get_item__ method to return speed and load of a selected dataset item
This way we can use additional mode information for our analysis

Reply:
Updated